routes/post.go

```diff
@@ -958,6 +963,129 @@ func (fes *APIServer) GetPostsStateless(ww http.ResponseWriter, req *http.Reques
+	PostsSkipped []*SkippedPostEntryResponse
+}
+
+func (fes *APIServer) GetPostsHashlist(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetPostsHashlistRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetPostsHashlist: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Decode the reader public key into bytes. Default to nil if no pub key is passed in.
+	var readerPublicKeyBytes []byte
+	if requestData.ReaderPublicKeyBase58Check != "" {
+		var err error
+		readerPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.ReaderPublicKeyBase58Check)
+		if requestData.ReaderPublicKeyBase58Check != "" && err != nil {
+			_AddBadRequestError(ww,
+				fmt.Sprintf("GetPostsHashlist: Problem decoding user public key: %v : %s", err, requestData.ReaderPublicKeyBase58Check))
+			return
+		}
+	}
+
+	// Get a view with all the mempool transactions.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetPostsHashlist: Error constucting utxoView: %v", err))
+		return
+	}
+
+	postEntryResponses := []*PostEntryResponse{}
+	skippedPostEntryResponses := []*SkippedPostEntryResponse{}
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetPostsHashlist: Error fetching blocked pub keys for user: %v", err))
+		return
+	}
+
+
+	for _, postHashHex := range requestData.PostsHashlist {
+		postHash, err := GetPostHashFromPostHashHex(postHashHex)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("GetPostsHashlist: %v", err))
+			return
+		}
+
+		// Fetch the postEntry requested.
+		postEntry := utxoView.GetPostEntryForPostHash(postHash)
+		if postEntry == nil {
+			// Post not found, add to skipped list
+			skippedPostEntryResponse := &SkippedPostEntryResponse{
+				PostHashHex: postHashHex,
+				Error: "GetPostsHashlist: Post not found",
+			}
+			skippedPostEntryResponses = append(skippedPostEntryResponses, skippedPostEntryResponse)
+			continue
+		}
+		var postEntryResponse *PostEntryResponse
+		postEntryResponse, err = fes._postEntryToResponse(postEntry, false, fes.Params, utxoView, readerPublicKeyBytes, 2)
+		if err != nil {
+			// Just skip posts that fail to convert for whatever reason.
+			skippedPostEntryResponse := &SkippedPostEntryResponse{
+				PostHashHex: postHashHex,
+				Error: fmt.Sprintf("GetPostsHashlist: Failed to convert post: %v", err),
+			}
+			skippedPostEntryResponses = append(skippedPostEntryResponses, skippedPostEntryResponse)
+			continue
+		}
+		profileEntry := utxoView.GetProfileEntryForPublicKey(postEntry.PosterPublicKey)
+		if profileEntry != nil {
+			// Convert it to a response since that sanitizes the inputs.
+			profileEntryResponse := fes._profileEntryToResponse(profileEntry, utxoView)
+			postEntryResponse.ProfileEntryResponse = profileEntryResponse
+		}
```

Comment:
we can create a little cache to keep track of the poster public keys we've seen and map them to profileEntryResponses to avoid calling `GetProfileEntryForPublicKey` and `_profileEntryToResponse` multiple times for the same public key.

Reply:
Yes, good idea, Implemented cache